Utilities/WDLs/DownsampleAndCollectCoverage.wdl

```diff
@@ -0,0 +1,172 @@
+version 1.0
+
+workflow DownsampleAndCollectCoverage {
+    input {
+        File input_cram
+        File input_cram_index
+        File ref_fasta
+        File ref_fasta_index
+
+        Float? target_coverage
+        Float? downsample_probability
+
+        File? coverage_intervals
+        String downsample_strategy = "ConstantMemory"
+        Int read_length = 150
+        Boolean use_fast_algorithm = true
+
+        String docker = "us.gcr.io/broad-gatk/gatk:4.4.0.0"
+        File? picard_jar_override
+
+        Int preemptible = 1
+    }
+
+    if (!defined(downsample_probability)) {
+        call CollectWgsMetrics as CollectOriginalCoverage {
+            input:
+                input_cram = input_cram,
+                input_cram_index = input_cram_index,
+                coverage_intervals = coverage_intervals,
+                ref_fasta = ref_fasta,
+                ref_fasta_index = ref_fasta_index,
+                read_length = read_length,
+                use_fast_algorithm = use_fast_algorithm,
+                docker = docker,
+                picard_jar_override = picard_jar_override,
+                preemptible = preemptible
+        }
+    }
+
+    call Downsample {
+        input:
+            input_cram = input_cram,
+            downsample_probability = downsample_probability,
+            target_coverage = target_coverage,
+            original_coverage = CollectOriginalCoverage.mean_coverage,
+            downsample_strategy = downsample_strategy,
+            ref_fasta = ref_fasta,
+            ref_fasta_index = ref_fasta_index,
+            docker = docker,
+            picard_jar_override = picard_jar_override,
+            preemptible = preemptible
+    }
+
+    call CollectWgsMetrics as CollectDownsampledCoverage {
+        input:
+            input_cram = Downsample.downsampled_cram,
+            input_cram_index = Downsample.downsampled_cram_index,
+            coverage_intervals = coverage_intervals,
+            ref_fasta = ref_fasta,
+            ref_fasta_index = ref_fasta_index,
+            read_length = read_length,
+            use_fast_algorithm = use_fast_algorithm,
+            docker = docker,
+            picard_jar_override = picard_jar_override,
+            preemptible = preemptible
+    }
+
+    output {
+        File downsampled_cram = Downsample.downsampled_cram
+        File downsampled_cram_index = Downsample.downsampled_cram_index
+        Float downsampled_mean_coverage = CollectDownsampledCoverage.mean_coverage
```

Comment:
Do you think it makes sense to also return the `wgs_metrics` files used to compute the mean coverage? Sometimes those other stats (like the variance of coverage, etc) are also useful to have in a Terra table.